src/main/java/bio/terra/service/iam/sam/SamIam.java

```diff
@@ -205,6 +205,21 @@ private Void deleteResourceInner(AuthenticatedUserRequest userReq,
 
         // create the resource in sam
         createResourceCorrectCall(samResourceApi.getApiClient(), IamResourceType.DATASET.toString(), req);
+        return null;
+    }
+
+    @Override
+    public Map<IamRole, String> syncDatasetResourcePolicies(AuthenticatedUserRequest userReq, UUID datasetId)
+        throws InterruptedException {
+        SamRetry samRetry = new SamRetry(configurationService);
```

Comment:
I realize `SamRetry` is existing code, but IMO the pattern: 
- `x = new X()`
- `x.someMethod()`
- `x` is never used again

Is better represented as a static method on `X`. Internally this method can still create an `X`.

Reply:
Ok - I think I've switched it over!

Reply:
Actually, per @nmalfroy 's comment, I think I'm going to revert these changes.

Reply:
Sorry, I think I left out an important part of this suggestion. I meant to make `retry` a static method on `SamRetry`, not to make a static instance of the `SamRetry` object. I made the same suggestion on a recent stairway PR, and Dan's change includes this suggestion and one that fixes the need for `Void` returns. See `DbRetry`:

https://github.com/DataBiosphere/stairway/blob/develop/src/main/java/bio/terra/stairway/DbRetry.java

Reply:
Ah, ok. I totally misunderstood. I think I've followed the dbRetry example. I'll take a look at method overloading.